crates/runtime/benches/setup/mod.rs

```diff
@@ -132,3 +117,104 @@ fn get_branch_name() -> String {
+            app_builder = app_builder
+                .with_dataset(self.make_rw_dataset(upload_results_dataset, "oss_benchmarks"));
+        }
+
+        app_builder.build()
+    }
+
+    fn make_dataset(&self, path: &str, name: &str) -> Dataset {
+        Dataset::new(format!("spiceai:{path}"), name.to_string())
+    }
+
+    fn make_rw_dataset(&self, path: &str, name: &str) -> Dataset {
+        let mut ds = Dataset::new(format!("spiceai:{path}"), name.to_string());
+        ds.mode = Mode::ReadWrite;
+        ds.replication = Some(Replication { enabled: true });
+        ds
+    }
+}
+
+struct PostgresBenchAppBuilder {}
+
+impl PostgresBenchAppBuilder {
+    fn get_postgres_params() -> Option<Params> {
+        let pg_host = std::env::var("PG_BENCHMARK_PG_HOST").unwrap();
+        let pg_user = std::env::var("PG_BENCHMARK_PG_USER").unwrap();
+        let pg_pass = std::env::var("PG_BENCHMARK_PG_PASS").unwrap();
+        let pg_db = std::env::var("PG_BENCHMARK_PG_DB").unwrap();
+        let pg_sslmode = std::env::var("PG_BENCHMARK_PG_SSLMODE").unwrap();
+        // Get postgres params from github secret?
+        Some(Params::from_string_map(
+            vec![
+                ("pg_host".to_string(), pg_host),
+                ("pg_user".to_string(), pg_user),
+                ("pg_db".to_string(), pg_db),
+                ("pg_pass".to_string(), pg_pass),
+                ("pg_sslmode".to_string(), pg_sslmode),
+            ]
+            .into_iter()
+            .collect(),
+        ))
+    }
+}
+
+impl BenchAppBuilder for PostgresBenchAppBuilder {
+    fn build_app(&self, upload_results_dataset: &Option<String>) -> App {
+        let mut app_builder = AppBuilder::new("runtime_benchmark_test")
+            .with_secret_store(SpiceSecretStore::File)
+            .with_dataset(self.make_dataset("customer", "customer"))
+            .with_dataset(self.make_dataset("lineitem", "lineitem"))
+            .with_dataset(self.make_dataset("part", "part"))
+            .with_dataset(self.make_dataset("partsupp", "partsupp"))
+            .with_dataset(self.make_dataset("orders", "orders"))
+            .with_dataset(self.make_dataset("nation", "nation"))
+            .with_dataset(self.make_dataset("region", "region"))
+            .with_dataset(self.make_dataset("supplier", "supplier"));
+
+        if let Some(upload_results_dataset) = upload_results_dataset {
+            app_builder = app_builder
+                .with_dataset(self.make_rw_dataset(upload_results_dataset, "oss_benchmarks"));
+        }
+
+        app_builder.build()
+    }
+
+    fn make_dataset(&self, path: &str, name: &str) -> Dataset {
+        let mut dataset = Dataset::new(format!("postgres:{path}"), name.to_string());
+        dataset.params = Self::get_postgres_params();
+        dataset
+    }
+
+    fn make_rw_dataset(&self, path: &str, name: &str) -> Dataset {
+        let mut dataset = Dataset::new(format!("postgres:{path}"), name.to_string());
```

Comment:
I think we should consolidate all bench results in single place, we should not use data connector to store results IMO. Should be spiceai.

Reply:
Resolved in new commit